lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -363,27 +365,245 @@
+
+        //If IsCallable(callbackfn) is false, throw a TypeError exception.
+        if (typeof callbackfn !== "function") {
+            __chakraLibrary.raiseFunctionArgument_NeedFunction("Array.prototype.some");
+        }
+
+        //If thisArg is present, let T be thisArg; else let T be undefined.
+        //Let k be 0.
+        let k = 0;
+
+        //Repeat, while k < len
+        while (k < len) {
+            //Let Pk be ! ToString(k).
+            //Let kPresent be ? HasProperty(O, Pk).
+            //If kPresent is true, then
+            if (k in o) {
+                //Let kValue be ? Get(O, Pk).
+                let kValue = o[k];
+                //Let testResult be ToBoolean(? Call(callbackfn, T, « kValue, k, O »)).
+                //If testResult is true, return true.
+                if (__chakraLibrary.builtInCallInstanceFunction(callbackfn, thisArg, kValue, k, o)) {
+                    return true;
+                }
+            }
+            //Increase k by 1.
+            k++;
+        }
+        //Return false. 
+        return false;
+    });
+
+    platform.registerFunction(platform.FunctionKind.Array_every, function (callbackfn, thisArg = undefined) {
+        // ECMAScript 2017 #sec-array.prototype.every
+        
+        //Let O be ? ToObject(this value).
+        //Let len be ? ToLength(? Get(O, "length")).
+        let objInfo = __chakraLibrary.CheckArrayAndGetLen(this, "Array.prototype.every");
+        let o = objInfo.o;
+        let len = objInfo.len;
+
+        //If IsCallable(callbackfn) is false, throw a TypeError exception.
+        if (typeof callbackfn !== "function") {
+            __chakraLibrary.raiseFunctionArgument_NeedFunction("Array.prototype.every");
+        }
+
+        //If thisArg is present, let T be thisArg; else let T be undefined.
+        //Let k be 0.
+        let k = 0;
+
+        //Repeat, while k < len
+        while (k < len) {
+            //Let Pk be ! ToString(k).
+            //Let kPresent be ? HasProperty(O, Pk).
+            //If kPresent is true, then
+            if (k in o) {
+                //Let kValue be ? Get(O, Pk).
+                let kValue = o[k];
+                //Let testResult be ToBoolean(? Call(callbackfn, T, « kValue, k, O »)).
+                //If testResult is false, return false.
+                if (!__chakraLibrary.builtInCallInstanceFunction(callbackfn, thisArg, kValue, k, o)) {
+                    return false;
+                }
+            }
+            //Increase k by 1.
+            k++;
+        }
+        //Return true. 
+        return true;
+    });
+
+    platform.registerFunction(platform.FunctionKind.Array_includes, function (searchElement, fromIndex = undefined) {
+        // ECMAScript 2017 #sec-array.prototype.includes
+
+        if (this === null || this === undefined) {
+            __chakraLibrary.raiseThis_NullOrUndefined("Array.prototype.includes");
+        }
+
+        //Let O be ? ToObject(this value).
```

Comment:
why not use ```CheckArrayAndGetLen``` here?

Reply:
and possibly change in the indexOf function above

---
In reply to: [219936874](https://github.com/Microsoft/ChakraCore/pull/5723#discussion_r219936874) [](ancestors = 219936874)

Reply:
7a249b412b8679e7157949b794aedc7b19fc7252